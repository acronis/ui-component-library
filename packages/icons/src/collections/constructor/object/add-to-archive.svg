<svg fill="none" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
  <path fill="currentColor" d="M2 3v5.5h1V21h18V8.5h1V3H2Zm17.5 16.5h-15v-11h15v11Zm1-12.5h-17V4.5h17V7Zm-6 7.542h-1.75V11h-1.5v3.542H9.5l2.5 2.5 2.5-2.5Z"/>
</svg>
