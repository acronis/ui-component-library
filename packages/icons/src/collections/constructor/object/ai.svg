<svg fill="none" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
  <path fill="currentColor" d="M5.411 9.697a3.796 3.796 0 0 1-1.064-2.02c.33-.252 1.62-.324 3.575.195.289-.186.582-.362.876-.538l.092-.056c.093-.056.186-.113.28-.168.175-.1.354-.195.533-.29l.2-.106-.005-.001.142-.078c.14-.077.279-.154.42-.227.53-1.955 1.239-3.037 1.623-3.197.661.276 2.293 3.267 2.293 8.789 0 .627-.024 1.216-.062 1.777.54.23 1.093.43 1.655.604.067-.766.107-1.56.107-2.381 0-5.797-1.788-10.497-3.993-10.497-1.41 0-2.645 1.928-3.354 4.83-2.869-.837-5.156-.73-5.861.49-.554.96-.028 2.42 1.268 4.003.4-.385.828-.761 1.275-1.13Zm6.672 11.092c-.383-.16-1.091-1.241-1.619-3.194a31.69 31.69 0 0 1-.904-.49l-.116-.063c-.09-.049-.177-.097-.264-.147a21.067 21.067 0 0 1-.497-.304l-.212-.132.001.005a17.18 17.18 0 0 1-.545-.335c-1.958.52-3.25.446-3.58.193-.092-.71 1.683-3.619 6.465-6.38a26.04 26.04 0 0 1 1.57-.835 17.454 17.454 0 0 0-.305-1.735c-.697.325-1.404.688-2.115 1.098-5.02 2.899-8.197 6.796-7.094 8.706.704 1.22 2.991 1.325 5.858.489.71 2.904 1.947 4.832 3.357 4.832 1.107 0 2.109-1.186 2.832-3.1a21.643 21.643 0 0 1-1.615-.54 3.798 3.798 0 0 1-1.217 1.932Zm4.865-15.012c2.02-.33 3.548-.056 4.101.902.705 1.222-.347 3.256-2.507 5.323 2.158 2.065 3.212 4.098 2.507 5.319-1.102 1.91-6.066 1.108-11.087-1.791-.71-.41-1.378-.842-2.008-1.283.43-.4.88-.777 1.35-1.131.467.313.966.628 1.508.942 4.782 2.76 8.188 2.844 8.758 2.41.053-.414-.53-1.57-1.958-3.005.008-.16.011-.32.015-.48l.003-.16-.003.003.007-.21c.007-.208.014-.415.016-.621 0-.113-.004-.227-.006-.34l-.002-.093a30.824 30.824 0 0 0-.029-1.03c1.428-1.433 2.01-2.586 1.957-2.999a3.799 3.799 0 0 0-2.28-.087 21.803 21.803 0 0 0-.341-1.67Z" clip-rule="evenodd"/>
</svg>
