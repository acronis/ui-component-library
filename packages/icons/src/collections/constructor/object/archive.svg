<svg fill="none" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
  <path fill="currentColor" d="M2 3v5.5h1V21h18V8.5h1V3H2Zm17.5 16.5h-15v-11h15v11Zm1-12.5h-17V4.5h17V7ZM9.75 12.75h4.5a.75.75 0 1 0 0-1.5h-4.5a.75.75 0 1 0 0 1.5Z"/>
</svg>
