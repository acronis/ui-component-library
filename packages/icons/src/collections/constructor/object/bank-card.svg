<svg fill="none" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
  <path fill="currentColor" d="M19 4H5a3 3 0 0 0-3 3v10a3 3 0 0 0 3 3h14a3 3 0 0 0 3-3V7a3 3 0 0 0-3-3Zm1.5 13a1.502 1.502 0 0 1-1.5 1.5H5A1.502 1.502 0 0 1 3.5 17v-6.5h17V17Zm-17-8V7A1.502 1.502 0 0 1 5 5.5h14A1.502 1.502 0 0 1 20.5 7v2h-17Z"/>
</svg>
