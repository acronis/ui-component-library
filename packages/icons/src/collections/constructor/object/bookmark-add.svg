<svg fill="none" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
  <path fill="currentColor" d="M18.5 20.139V3.5h-13v16.639l6.5-3.715 6.5 3.715ZM4 2h16v20.724l-8-4.573-8 4.573V2Zm8.75 5v2.75h2.75v1.5h-2.75V14h-1.5v-2.75H8.5v-1.5h2.75V7h1.5Z" clip-rule="evenodd"/>
</svg>
