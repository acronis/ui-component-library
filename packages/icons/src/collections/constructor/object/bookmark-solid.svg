<svg fill="none" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
  <path fill="currentColor" d="M20 2H4v20.724l8-4.573 8 4.573V2Z"/>
</svg>
