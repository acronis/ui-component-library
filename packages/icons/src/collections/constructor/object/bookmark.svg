<svg fill="none" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
  <path fill="currentColor" d="M18.5 3.5v16.639l-5.756-3.29-.744-.425-.744.425L5.5 20.14V3.5h13ZM20 2H4v20.724l8-4.573 8 4.573V2Z"/>
</svg>
