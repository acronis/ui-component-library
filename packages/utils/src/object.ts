export function omit<Obj extends object, Keys extends keyof Obj>(
  obj: Obj,
  keys: Keys[],
): Omit<Obj, Keys> {
  const result = { ...obj };

  for (const key of keys) {
    delete result[key];
  }

  return result;
}

export function pick<Obj extends object, Keys extends keyof Obj>(
  obj: Obj,
  keys: Keys[],
): Pick<Obj, Keys> {
  const result = {} as Pick<Obj, Keys>;

  for (const key of keys) {
    if (key in obj) {
      result[key] = obj[key];
    }
  }

  return result;
}

/**
 * Parse jsonString and in case of exception return defaultValue
 */
export function parseJsonSafe<T>(jsonString: string, defaultValue: T): T;
export function parseJsonSafe<T>(jsonString: string): T | undefined;
export function parseJsonSafe<T>(jsonString: string, defaultValue?: T) {
  try {
    return JSON.parse(jsonString);
  }
  catch (e) {
    return defaultValue;
  }
}
