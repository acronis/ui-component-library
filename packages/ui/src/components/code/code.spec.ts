import { describe, expect, it } from 'vitest';
import { mount } from '@vue/test-utils';
import Code from './code.vue';
import type { CodeProps } from './code.ts';

const code = `from pymatgen.core import Lattice, Structure
from rolos_sdk import Dataframe, DataStorageInterface, DataStorageType, RawObject, TableColumn
from rolos_sdk.structures.object.pymatgen import PyMatGenObject

table_schema = [
    TableColumn(name="Index", type=int),
    TableColumn(name="Test", type=float),
    TableColumn(name="Structure", type=PyMatGenObject),
    TableColumn(name="Raw experiment data", type=RawObject),
]`;

describe('test Code component', () => {
  it('default props', () => {
    const wrapper = mount(Code);
    expect(wrapper.props()).toMatchInlineSnapshot(`
      {
        "code": undefined,
        "extension": undefined,
      }
    `);
  });

  it('pass props', () => {
    const wrapper = mount(Code, {
      props: {
        title: 'test',
      } as CodeProps,
    });

    expect(wrapper.props()).toMatchInlineSnapshot(`
      {
        "code": undefined,
        "extension": undefined,
      }
    `);
  });

  it('renders', () => {
    const wrapper = mount(Code);
    expect(wrapper.html()).toMatchInlineSnapshot(`"<code data-v-28437d1d="" class="acv-code"></code>"`);
  });

  it('code should be highlighted', () => {
    const wrapper = mount(Code, {
      props: { code, extension: 'py' }
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
