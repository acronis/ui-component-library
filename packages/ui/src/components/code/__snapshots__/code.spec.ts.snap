// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test Code component > code should be highlighted 1`] = `
<code
  class="acv-code"
  data-v-28437d1d=""
>
  <span
    class="hljs-keyword"
  >
    from
  </span>
   pymatgen.core 
  <span
    class="hljs-keyword"
  >
    import
  </span>
   Lattice, Structure

  <span
    class="hljs-keyword"
  >
    from
  </span>
   rolos_sdk 
  <span
    class="hljs-keyword"
  >
    import
  </span>
   Dataframe, DataStorageInterface, DataStorageType, RawObject, TableColumn

  <span
    class="hljs-keyword"
  >
    from
  </span>
   rolos_sdk.structures.
  <span
    class="hljs-built_in"
  >
    object
  </span>
  .pymatgen 
  <span
    class="hljs-keyword"
  >
    import
  </span>
   PyMatGenObject

table_schema = [
    TableColumn(name=
  <span
    class="hljs-string"
  >
    "Index"
  </span>
  , 
  <span
    class="hljs-built_in"
  >
    type
  </span>
  =
  <span
    class="hljs-built_in"
  >
    int
  </span>
  ),
    TableColumn(name=
  <span
    class="hljs-string"
  >
    "Test"
  </span>
  , 
  <span
    class="hljs-built_in"
  >
    type
  </span>
  =
  <span
    class="hljs-built_in"
  >
    float
  </span>
  ),
    TableColumn(name=
  <span
    class="hljs-string"
  >
    "Structure"
  </span>
  , 
  <span
    class="hljs-built_in"
  >
    type
  </span>
  =PyMatGenObject),
    TableColumn(name=
  <span
    class="hljs-string"
  >
    "Raw experiment data"
  </span>
  , 
  <span
    class="hljs-built_in"
  >
    type
  </span>
  =RawObject),
]
</code>
`;
